api/src/controllers/document-controller.js

```diff
@@ -15,33 +16,40 @@ const s3 = new AWS.S3({
   s3ForcePathStyle: true
 });
 
-exports.protectedOptions = function (args, res, next) {
+exports.protectedOptions = function(args, res, next) {
   res.status(200).send();
 };
 
-exports.protectedPost = async function (args, res, next) {
+exports.protectedPost = async function(args, res, next) {
   if (
     args.swagger.params.fileName &&
     args.swagger.params.fileName.value &&
     args.swagger.params.recordId &&
     args.swagger.params.recordId.value
   ) {
+    // Determine if the record is anonymous, and therefore the document must not be publicly available.
+    let isAnonymous = false;
+    const db = mongodb.connection.db(process.env.MONGODB_DATABASE || 'nrpti-dev');
+    const collection = db.collection('nrpti');
+    const masterRecord = await collection.findOne({ _id: new ObjectID(args.swagger.params.recordId.value) });
+    if (Object.prototype.hasOwnProperty.call(mongoose.model(masterRecord._schemaName).schema.obj, 'issuedTo')) {
+      // Only check for anonymity if the master schema has entity information.  Otherwise assume not anonymous.
+      isAnonymous = queryUtils.isRecordAnonymous(masterRecord);
+    }
+
```

Comment:
It was using a db connection like this before, I just moved it to the start of the function since I added another place it gets used down below.  But, I think you're right, so Ill update it here and below.

Reply:
Actually, I don't think we can change this to just mongoose.

We only know the incoming recordId, we don't know what the model is.  And mongoose always requires that you pick a model, I believe?